hasjob/models/filterset.py

```diff
@@ -0,0 +1,174 @@
+)
+
+
+filterset_jobcategory_table = db.Table('filterset_jobcategory', db.Model.metadata,
+    db.Column('filterset_id', None, db.ForeignKey('filterset.id'), primary_key=True),
+    db.Column('jobcategory_id', None, db.ForeignKey('jobcategory.id'), primary_key=True, index=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+
+filterset_tag_table = db.Table('filterset_tag', db.Model.metadata,
+    db.Column('filterset_id', None, db.ForeignKey('filterset.id'), primary_key=True),
+    db.Column('tag_id', None, db.ForeignKey('tag.id'), primary_key=True, index=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+filterset_domain_table = db.Table('filterset_domain', db.Model.metadata,
+    db.Column('filterset_id', None, db.ForeignKey('filterset.id'), primary_key=True),
+    db.Column('domain_id', None, db.ForeignKey('domain.id'), primary_key=True, index=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+
+class FilterSet(BaseScopedNameMixin, db.Model):
+    """
+    Store filters to display a filtered set of jobs scoped by a board on SEO friendly URLs
+
+    Eg: `https://hasjob.co/f/machine-learning-jobs-in-bangalore`
+    """
+
+    __tablename__ = 'filterset'
+
+    board_id = db.Column(None, db.ForeignKey('board.id'), nullable=False, index=True)
+    board = db.relationship(Board)
+    parent = db.synonym('board')
+
+    #: Welcome text
+    description = db.Column(db.UnicodeText, nullable=False, default=u'')
+
+    #: Associated job types
+    types = db.relationship(JobType, secondary=filterset_jobtype_table)
+    #: Associated job categories
+    categories = db.relationship(JobCategory, secondary=filterset_jobcategory_table)
+    tags = db.relationship(Tag, secondary=filterset_tag_table)
+    domains = db.relationship(Domain, secondary=filterset_domain_table)
+    location_geonameids = db.Column(postgresql.ARRAY(db.Integer(), dimensions=1), default=[], nullable=False, index=True)
+    remote_location = db.Column(db.Boolean, default=False, nullable=False, index=True)
+    pay_currency = db.Column(db.CHAR(3), nullable=True, index=True)
+    pay_cash = db.Column(db.Integer, nullable=True, index=True)
+    equity = db.Column(db.Boolean, nullable=False, default=False, index=True)
+    keywords = db.Column(db.Unicode(250), nullable=False, default=u'', index=True)
+
+    def __repr__(self):
+        return '<FilterSet %s "%s">' % (self.board.title, self.title)
+
+    @classmethod
+    def get(cls, board, name):
+        return cls.query.filter(cls.board == board, cls.name == name).one_or_none()
+
+    def url_for(self, action='view', _external=False, **kwargs):
+        if action == 'view':
+            subdomain = self.board.name if self.board.not_root else None
+            return url_for('filterset_view', subdomain=subdomain, name=self.name, _external=_external)
```

Comment:
I have an idea for making this work with `is_url_for`:

```python
def url_for(self, action='view', _external=True, **kwargs):
    kwargs.setdefault('subdomain', self.board.name if self.board.not_root else None)
    return super(FilterSet, self).url_for(action, kwargs)
```

Reply:
Also made external True since that's the only condition under which this will be used. Not sold on it though.

Reply:
(Forgot to pass it in super)

Reply:
`filterset.url_for('view', **query_params)` in `index.html.jinja2` fails with this though, since more than 2 arguments are supplied.

Reply:
You're accepting `**kwargs`. Why does it fail?